pkg/comp-functions/functions/vshnforgejo/deploy.go

```diff
@@ -0,0 +1,204 @@
+	"github.com/vshn/appcat/v4/pkg/comp-functions/runtime"
+)
+
+// DeployForgejo deploys a Forgejo instance via the Helm Chart.
+func DeployForgejo(ctx context.Context, comp *vshnv1.VSHNForgejo, svc *runtime.ServiceRuntime) *xfnproto.Result {
+
+	err := svc.GetObservedComposite(comp)
+	if err != nil {
+		return runtime.NewFatalResult(fmt.Errorf("cannot get composite: %w", err))
+	}
+
+	svc.Log.Info("Bootstrapping instance namespace and rbac rules!")
+	err = common.BootstrapInstanceNs(ctx, comp, comp.GetServiceName(), comp.GetName()+"-instanceNs", svc)
+	if err != nil {
+		return runtime.NewWarningResult(fmt.Sprintf("cannot bootstrap instance namespace: %s", err))
+	}
+
+	secretName, err := common.AddCredentialsSecret(comp, svc, []string{"password"}, common.AddStaticFieldToSecret(map[string]string{
+		"username": "gitea_admin",
+	}))
+	if err != nil {
+		return runtime.NewWarningResult("cannot add credentials secret")
+	}
+
+	connDetails, err := svc.GetObservedComposedResourceConnectionDetails(secretName)
+	if err != nil {
+		return runtime.NewWarningResult("cannot get connection details")
+	}
+
+	svc.SetConnectionDetail("FORGEJO_USERNAME", connDetails["username"])
+	svc.SetConnectionDetail("FORGEJO_PASSWORD", connDetails["password"])
+	// "," as separator for multiple FQDNs, it's better than anything else as highlighting with mouse in terminal works well
+	// result is fqdn[0],fqdn[1],fqdn[2]
+	svc.SetConnectionDetail("FORGEJO_URL", []byte(strings.Join(comp.Spec.Parameters.Service.FQDN, ",")))
+
+	svc.Log.Info("Adding forgejo release")
+	err = addForgejo(ctx, svc, comp, secretName)
+	if err != nil {
+		return runtime.NewWarningResult(fmt.Sprintf("cannot add forgejo release: %s", err))
+	}
+
+	return nil
+}
+
+func addForgejo(ctx context.Context, svc *runtime.ServiceRuntime, comp *vshnv1.VSHNForgejo, secretName string) error {
+
+	securityContext := map[string]any{}
+	if svc.Config.Data["isOpenshift"] == "true" {
+		securityContext = map[string]any{
+			"runAsUser":                nil,
+			"allowPrivilegeEscalation": false,
+			"podSecurityContext": map[string]any{
+				"fsGroup": nil,
+			},
+			"containerSecurityContext": map[string]any{
+				"runAsUser":                nil,
+				"allowPrivilegeEscalation": false,
+			},
+			"capabilities": map[string]any{
+				"drop": []string{
+					"ALL",
+				},
+			},
+		}
+	}
+
+	values := map[string]any{
+		"gitea": map[string]any{
+			"admin": map[string]any{
+				"existingSecret": secretName,
+			},
+			"config": map[string]any{
+				"admin": map[string]any{
+					"SEND_NOTIFICATION_EMAIL_ON_NEW_USER": true,
+				},
+				"cache": map[string]any{
+					"ADAPTER": "twoqueue",
+					"HOST":    "'{\"size\":100, \"recent_ratio\":0.25, \"ghost_ratio\":0.5}'",
+				},
+				"cron": map[string]any{
+					"ENABLED": true,
+					"archive_cleanup": map[string]any{
+						"SCHEDULE":   "@hourly",
+						"OLDER_THAN": "2h",
+					},
+				},
+				"database": map[string]any{
```

Comment:
I'd suggest to set sqlite to WAL mode. It will increase the sqlite performance quite a bit: https://forgejo.org/docs/latest/admin/database-preparation/#sqlite

```
"SQLITE_JOURNAL_MODE": "WAL",
```

Reply:
1. OCI image refused to be found without that tweak, I just saw that it's similarily solved in provider-helm/examples example
2. added that